sklearn/utils/estimator_checks.py

```diff
@@ -331,6 +332,44 @@ def _construct_instance(Estimator):
     return estimator
 
 
+def _get_xfail_checks(estimator):
+    """Get xfail_check from estimator"""
+    if isinstance(estimator, type):
+        # try to construct estimator instance, if it is unable to then
+        # return then ignore xfail tag
+        try:
+            estimator = _construct_instance(estimator)
+        except Exception:
+            return {}
+
+    if not hasattr(estimator, "_get_tags"):
+        return {}
```

Comment:
why is this needed? estimators inheriting from BaseEstimator should have a `_get_tags` method

Reply:
The following

https://github.com/scikit-learn/scikit-learn/blob/2587a033808506cd1e46b0cba18fae5ae829f4ab/sklearn/utils/tests/test_estimator_checks.py#L365

would fail at the incorrect place.

Reply:
I think we should just change the test then. Which is what I did for the one just below. The previous error message wasn't that great either anyway